src/table/virtualized-table/hooks/use-data/index.ts

```diff
@@ -109,11 +111,12 @@ const useData = (
     const qWidth = Math.min(100, layout.qHyperCube.qSize.qcx, visibleColumnCount + COLUMN_DATA_BUFFER_SIZE);
     const qHeight = Math.min(100, layout.qHyperCube.qSize.qcy, visibleRowCount + ROW_DATA_BUFFER_SIZE);
 
-    getDataPages([{ qLeft: 0, qTop, qHeight, qWidth }]).then((dataPages) => {
+    const onBeforeHandlePages = () => {
       setRowsInPage(Array(rowCount).fill(undefined)); // Reset rows to initial value
-      handleDataPages(dataPages);
-    });
-  }, [getDataPages, handleDataPages, layout, visibleRowCount, visibleColumnCount, pageInfo, queue, rowCount]);
+    };
+
+    queue.enqueue({ qLeft: 0, qTop, qHeight, qWidth }, onBeforeHandlePages);
```

Comment:
The change in this PR can result in 2-3 renders in the `Table` component that would all request new data with possibly slightly different call parameters as qWidth and/or qHeight might change. Potentially triggering a race condition.

So I changed to use the queue her for fetching data as the `clear()` call in this useEffect will cancel any ongoing requests.